udf-compiler/pom.xml

```diff
@@ -141,6 +134,13 @@
                     <groupId>org.apache.curator</groupId>
                     <artifactId>curator-recipes</artifactId>
                     <version>4.3.0.7.2.7.0-184</version>
+                    <scope>provided</scope>
+                </dependency>
+                <dependency>
+                    <groupId>com.google.guava</groupId>
+                    <artifactId>guava</artifactId>
+                    <version>28.0-jre</version>
+                    <scope>provided</scope>
```

Comment:
Guava is notorious for being incompatible between the versions.  If the UDF compiled needs Guava we'll probably need to shade it to guarantee it gets the version it needs rather than hope the one it finds in the Spark env is compatible.

Reply:
changed to test scope and commented, asking Cloudera about it

Reply:
note it was just unit tests that needed it (thus the change to test scope - which is what I should have had it at in th efirst place)